Mathlib/NumberTheory/NumberField/House.lean

```diff
@@ -0,0 +1,310 @@
+/-
+Copyright (c) 2024 Michail Karatarakis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michail Karatarakis
+-/
+import Mathlib.NumberTheory.SiegelsLemma
+import Mathlib.NumberTheory.NumberField.CanonicalEmbedding.Basic
+import Mathlib.NumberTheory.NumberField.EquivReindex
+
+/-!
+
+# House of an algebraic number
+This file defines the house of an algebraic number `α`, which is
+the largest modulus of its conjugates.
+
+## References
+* [D. Marcus, *Number Fields*][marcus1977number]
+* [Keng, H. L, *Introduction to number theory*][keng1982house]
+
+## Tagshouse
+number field, algebraic number, house
+-/
+
+variable {K : Type*} [Field K] [NumberField K]
+
+namespace NumberField
+
+noncomputable section
+
+open Module.Free FiniteDimensional canonicalEmbedding Matrix Finset
+
+attribute [local instance] Matrix.seminormedAddCommGroup
+
+/-- The house of an algebraic number as the norm of its image by the canonical embedding.-/
```

Comment:
You should probably mentioned here also --as you did in the introduction--,  that the house is the largest of the modulus of the conjugates and refer to `house_eq_sup'` for the proof.